src/main/java/com/internet/shop/dao/jdbc/ShoppingCartJdbcImpl.java

```diff
@@ -0,0 +1,182 @@
+                ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+                if (generatedKeys.next()) {
+                    cart.setId(generatedKeys.getLong(1));
+                }
+
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("ShoppingCart for userId=" + cart.getUserId()
+                    + " wasn't created.", e);
+        }
+        return cart;
+    }
+
+    @Override
+    public Optional<ShoppingCart> get(Long id) {
+        String query = "SELECT * FROM shopping_carts WHERE cart_id = ? AND cart_deleted = FALSE;";
+        ShoppingCart shoppingCart = new ShoppingCart();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            try (PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+                preparedStatement.setLong(1, id);
+                ResultSet resultSet = preparedStatement.executeQuery();
+                if (resultSet.next()) {
+                    shoppingCart = getCartFromResultSet(resultSet);
+                }
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("ShoppingCart id=" + id + " wasn't found.", e);
+        }
+        shoppingCart.setProducts(getProductsFromCart(shoppingCart.getId()));
+        return Optional.of(shoppingCart);
+    }
+
+    @Override
+    public List<ShoppingCart> getAll() {
+        String query = "SELECT cart_id, user_id "
+                + "FROM shopping_carts "
+                + "WHERE cart_deleted = FALSE;";
+        List<ShoppingCart> shoppingCarts = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            try (PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+                ResultSet resultSet = preparedStatement.executeQuery();
+                while (resultSet.next()) {
+                    shoppingCarts.add(getCartFromResultSet(resultSet));
+                }
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get all ShoppingCarts", e);
+        }
+        for (ShoppingCart cart : shoppingCarts) {
+            cart.setProducts(getProductsFromCart(cart.getId()));
+        }
+        return shoppingCarts;
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart cart) {
+        deleteProductsFromCart(cart);
+        String query = "INSERT INTO shopping_carts_products (cart_id, product_id) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            for (Product product : cart.getProducts()) {
+                try (PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+                    preparedStatement.setLong(1, cart.getId());
+                    preparedStatement.setLong(2, product.getId());
+                    preparedStatement.executeUpdate();
+                }
+            }
```

Comment:
it's better to move this  to separate method and use here and in create()

Reply:
But when we are creating the cart we do not insert any product into it. We only create an empty cart and we do set the id to cart that's it.

Reply:
By the way, I pushed HW on Friday morning before some people :) and their hw was approved, it is unfair :)